### ISSUE TITLE
ITestOutputHelper  does not work in constructors/dispose or Teardown() when test-class contains xbehave-Scenarios

### ISSUE THREAD

**Issue by Jazzman82** [adamralph/xbehave.net#565]
Is it possible to add the ITestOutputHelper to those artifacts? It always says that there is no active test-case, but it should be possible as constructor/dispose and Teardown are test-case specific.

**Comment by adamralph** [comment]
Hi @Jazzman82 thanks for raising this. I've successfully reproduced the problem.

The good news is that I currently have some [work in progess](https://github.com/xbehave/xbehave.net/pull/566) which should fix this.

**Comment by Jazzman82** [comment]
Cool! So I will switch to patience-mode ;)

**Comment by redoz** [comment]
@adamralph Just ran into this myself, as I re-used a helper-method that writes to the `ITestOutputHelper` in a Teardown method, so I just wanted to check what the expectations are on this issue? Anything I can do to help out?

**Comment by Andras-Csanyi** [comment]
I have the same issue. Any progress on this?

**Comment by adamralph** [comment]
@SayusiAndo unfortunately I have had no time to work on this.

**Comment by Andras-Csanyi** [comment]
@adamralph No worries. I could solve it, unfortunately I had to drop xBehave.

**Comment by Eklange** [comment]
Any updates on this issue, would be great to be able to use ITestOutputHelper with xBehave

**Comment by adamralph** [comment]
I'm not aware of any progress since my last comment.

> Anything I can do to help out?

@redoz PR's are always welcome. 🙂

**Comment by redoz** [comment]
@adamralph Is #566 on hold waiting for xunit3?

**Comment by adamralph** [comment]
> Is #566 on hold waiting for xunit3?

@redoz no, it's not waiting for xunit3. It's something I started a while back, but didn't have time progress with any further.

That said, I'm not entirely sure if it's worth trying to get that all done (I'm not entirely sure where it's leading TBH) or whether it's worth just leaving xbehave 2.x as-is and waiting for xunit 3. Because xunit 2 was so different to xunit 1, xbehave 2 was a complete re-write, so it may be the same for xbehave 3.

**Comment by redoz** [comment]
@adamralph Yeah my impression was that doing significant work with xunit2 internals would be wasted effort as soon as xunit3 is released, so I'm looking for quick wins, and will see if the new version offers what I need before starting to extend it.
